arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -157,14 +174,42 @@ DBServerAgencySyncResult DBServerAgencySync::execute() {
   }
 
   auto& clusterInfo = _server.getFeature<ClusterFeature>().clusterInfo();
-  uint64_t planIndex = 0;
-  auto plan = clusterInfo.getPlan(planIndex);
+  uint64_t planIndex = 0, currentIndex = 0;
+
+
+  // The rationale is that even with 1000 databases and us waking up every 5 seconds,
+  // we should approximately visit every database without being dirty once per hour.
+  // That is why we divide by 720.
+  auto moreDirt = mfeature.pickRandomDirty(
+    static_cast<size_t>(std::ceil(static_cast<double>(mfeature.lastNumberOfDatabases()) / 720.)));
+  auto dirty = mfeature.dirty();
+  // Add `moreDirt` to `dirty` but remove from `moreDirt`, if already dirty anyway.
+  // Then we can reasonably be surprised if we find anything in a database in `moreDirt`.
+  for (auto it = moreDirt.begin(); it != moreDirt.end();) {
+    if (dirty.find(*it) != dirty.end()) {
+      dirty.insert(*it);
+      it++;
+    } else {
```

Comment:
This part confuses me. We are looking up `*it` in `dirty`, and if it is already in there, we insert it into... `dirty`??

Reply:
nice catch. many thx.